### ISSUE TITLE
About APs

### ISSUE THREAD

**Issue by paive** [ricvolpi/adversarial_feature_augmentation#6]
I trained a encoder $E_s$ and classifier $C$ on mnist with the final accuracy about 98%, and I computed the $APs$ using the method in the paper. 

1. The $APs$ of the features obtained using $E_s$ is nearly 60000 which is the size of training data.
2. Besides, I trained a feature generator $S$ using $E_s$ and CGAN. After finishing the training, I feed the $S$ with 60000 noises. However the $APs$ of the features generated is about 1800.

That is to say the above results is in contradiction with the results in the paper. But I have not found any bugs in my code, I want to learn that have you ever got such results.

Thanks!

**Comment by ricvolpi** [comment]
Hi, 

1. It's reasonable. Please note that in Table 1, MNIST row we report APs associated with the P1 experiment, where only 2,000 MNIST samples are used. 

2. We evaluate the number of APs that S is able to generate by feeding it with noise samples from the uniform distribution until the number of different APs saturates, and I would suggest you to evaluate your generator in the same way. However, 1,800 APs over 60,000 noise samples doesn't sound good. Is the minimax game defined to train S reaching a proper equilibrium? t-SNE plots like the one we report in Figure 2 can be helpful indicators.
